zio-connect/shared/src/main/scala/zio/connect/file/testkit/TestFileConnector.scala

```diff
@@ -0,0 +1,326 @@
+
+  override def deleteRecursivelyPath(implicit trace: Trace): ZSink[Any, IOException, Path, Nothing, Unit] =
+    ZSink.foreach(path => fs.deleteRecursively(path))
+
+  override def existsPath(path: Path)(implicit trace: Trace): ZSink[Any, IOException, Any, Nothing, Boolean] =
+    ZSink.fromZIO(fs.exists(path))
+
+  override def listPath(path: => Path)(implicit trace: Trace): ZStream[Any, IOException, Path] =
+    ZStream.unwrap(fs.list(path).map(a => ZStream.fromChunk(a)))
+
+  override def movePathZIO(
+    locator: Path => ZIO[Any, IOException, Path]
+  )(implicit trace: Trace): ZSink[Any, IOException, Path, Nothing, Unit] =
+    ZSink.foreach { oPath =>
+      fs.movePath(oPath, locator(oPath))
+    }
+
+  override def readPath(path: => Path)(implicit trace: Trace): ZStream[Any, IOException, Byte] =
+    ZStream.unwrap(fs.getContent(path).map(a => ZStream.fromChunk(a)))
+
+  override def tailPath(path: => Path, freq: => Duration)(implicit trace: Trace): ZStream[Any, IOException, Byte] = {
+
+    def push(index: Ref[Int], queue: Queue[Byte]): ZIO[Any, IOException, Unit] =
+      for {
+        content   <- fs.getContent(path)
+        idx       <- index.get
+        newContent = content.drop(idx)
+        _         <- queue.offerAll(newContent)
+        newIndex   = content.length
+        _         <- index.set(newIndex)
+
+      } yield ()
+
+    ZStream.unwrap(
+      for {
+        queue <- Queue.unbounded[Byte]
+        index <- Ref.make(0)
+        _ <-
+          push(index, queue).repeat[Any, (Long, Long)](Schedule.recurs(10) && Schedule.spaced(Duration.fromMillis(100)))
+      } yield ZStream.fromQueue(queue)
+    )
+
+  }
+
+  override def tailPathUsingWatchService(path: => Path, freq: => Duration)(implicit
+    trace: Trace
+  ): ZStream[Any, IOException, Byte] = tailPath(path, freq)
+
+  override def tempPath(implicit trace: Trace): ZSink[Scope, IOException, Any, Nothing, Path] =
+    ZSink.fromZIO(fs.tempPath)
+
+  override def tempPathIn(dirPath: Path)(implicit trace: Trace): ZSink[Scope, IOException, Any, Nothing, Path] =
+    ZSink.fromZIO(fs.tempPathIn(dirPath))
+
+  override def tempDirPath(implicit trace: Trace): ZSink[Scope, IOException, Any, Nothing, Path] =
+    ZSink.fromZIO(fs.tempDirPath)
+
+  override def tempDirPathIn(dirPath: Path)(implicit trace: Trace): ZSink[Scope, IOException, Any, Nothing, Path] =
+    ZSink.fromZIO(fs.tempDirPathIn(dirPath))
+
+  override def writePath(path: => Path)(implicit trace: Trace): ZSink[Any, IOException, Byte, Nothing, Unit] =
+    for {
+      _ <- ZSink.fromZIO(fs.removeContentIfExists(path))
+      _ <- ZSink.foreachChunk(bytes => fs.write(path, bytes))
+    } yield ()
+
+}
+
+object TestFileConnector {
+
+  def layer: ZLayer[Any, Nothing, FileConnector] = ZLayer.fromZIO(
+    STM.atomically {
+      for {
+        a <- TRef.make(Map.empty[Path, FileSystemNode])
+      } yield TestFileConnector(TestFileSystem(a))
+    }
+  )
+
+  private[testkit] sealed trait FileSystemNode {
```

Comment:
Moved all test structures in the TestFileConnector companion object and made them private.